core/modules/system/system.admin.inc

```diff
@@ -508,6 +508,7 @@ function system_modules($form, $form_state = array()) {
       'autocomplete' => 'off',
       'title' => t('Enter a part of the module name or description to filter.'),
     ),
+    '#default_value' => (backdrop_get_query_parameters()) ? backdrop_get_query_parameters() : '',
```

Comment:
This gets *all* parameters and puts them in the field. I tried a query string such as `?search=foo&example=bar` and "foo bar" ended up in the filter box.

Just checking `$_GET['filter']` would be a better way to prepopulate. (Also I think sticking with "filter" instead of "search" would be good for consistency).